dlt/destinations/filesystem/filesystem.py

```diff
@@ -82,17 +88,31 @@ def dataset_path(self) -> str:
     def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
         # clean up existing files for tables selected for truncating
         if truncate_tables and self.fs_client.isdir(self.dataset_path):
-            all_files = self.fs_client.ls(self.dataset_path, detail=False, refresh=True)
+
+            # collect files
+            all_files = []
+            for basedir, _dirs, files  in self.fs_client.walk(self.dataset_path, detail=False, refresh=True):
+                all_files += [posixpath.join(basedir, file) for file in files]
+
             for table in truncate_tables:
-                search_prefix = posixpath.join(self.dataset_path, f"{self.schema.name}.{table}.")
-                for item in all_files:
-                    # NOTE: glob implementation in fsspec does not look thread safe, way better is to use ls and then filter
-                    if item.startswith(search_prefix):
-                        # NOTE: deleting in chunks on s3 does not raise on access denied, file non existing and probably other errors
-                        self.fs_client.rm_file(item)
-
-        # create destination dir
-        self.fs_client.makedirs(self.dataset_path, exist_ok=True)
+                try:
+                    table_prefix = path_utils.get_table_prefix(self.config.layout, self.schema.name, table)
+                    search_prefix = posixpath.join(self.dataset_path, table_prefix)
+                    for item in all_files:
+                        # NOTE: glob implementation in fsspec does not look thread safe, way better is to use ls and then filter
+                        if item.startswith(search_prefix):
+                            # NOTE: deleting in chunks on s3 does not raise on access denied, file non existing and probably other errors
+                            self.fs_client.rm_file(item)
+                except CantExtractTablePrefix:
+                    # crazy deletions might happen if table files are not clearly separated, so do nothing
+                    # TODO: print a warning to console?
+                    pass
+
+        # create destination dirs for all tables
+        for tschema in self.schema.tables.values():
+            table_prefix = path_utils.get_table_prefix(self.config.layout, self.schema.name, tschema["name"])
+            destination_dir = posixpath.join(self.dataset_path, table_prefix)
+            self.fs_client.makedirs(destination_dir, exist_ok=True)
```

Comment:
does it really work? so if I have {data_set}/{schema_name}.{table_name} it will just create {data_set}? this code seems to work only with `/` separator where each table has a separate folder. what you should to here:
1. collect all paths for table_prefixes. remove the file name component so we have only folders
2. find unique paths
3. create those paths

so in the above case, even if you have 100 tables but all are in {dataset_name} folder, the code will create just one path

this also needs to be tested with a few layouts